### PR TITLE
Add a customFuncMap parameter

### PR DESCRIPTION
**Pull request by itdove**
I added that capabability because I needed for the cm_cli to overload the template functions.

Signed-off-by: Dominique Vernier <dvernier@redhat.com>

### CODE REVIEW

File: pkg/helpers/apply/apply.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ var (
 func ApplyDeployments(
 	kubeClient kubernetes.Interface,
 	reader asset.ScenarioReader,
+	customFuncMap template.FuncMap,
```
Comment (qiujian16) [inline pkg/helpers/apply/apply.go:48]: I am not sure the purpose of this. Is it to render the yaml template in a customized way?

**Comment by itdove** [comment]
It is to be able to add new functions in the Gand text/template rendering. For the time being there is the standard golang text functions plus spring, plus few others I implemented in the templatefuntion.go file. I want to implement few extra for the cm-cli but as they are specific to the cm cli I don't want to implement them in the clusteradm. This change will allow me to inject the extra functions I need for the cm cli without having them in the clusteradm

File: pkg/helpers/apply/applierBuilder.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	templateFuncMap     template.FuncMap
+}
+
+//ApplierBuilder a builder to build the applier
+type ApplierBuilder struct {
+	Applier
+}
+
+type iApplierBuilder interface {
+	Build() Applier
```
Comment (qiujian16) [inline pkg/helpers/apply/applierBuilder.go:26]: Add some comments for each method here.

**Comment by qiujian16** [comment]
it looks good. Would you squash the commit.

**Pull request closed without merging**